In [33]:
quant = int(input("Quantos trequinhos:"))
preço = float(input("preço dos trequinhos:"))

print("você comprou",quant,"trequinhos por R$",preço,"cada um")

você comprou 501 trequinhos por R$ 0.01 cada um


Agora vamos calcular o total

In [34]:
total = quant * preço
print("O total deu R$",total)

O total deu R$ 5.01


Para compras acima de R$ 1000 ou mais de 500 unidades dar 10% de desconto
(se comprar 501 trequinhos de R$ 0,01 ainda assim ganha desconto)

In [35]:
if (total >= 1000) or (quant > 500):
    desconto = total * (10/100) # 10%
    print(f"você ganhou R$ {desconto:.2f} de desconto")
    # recalcula o total
    total = total - desconto
    print(f"O total com desconto é R$ {total:.2f}")
else:
    print("não ganha desconto")

você ganhou R$ 0.50 de desconto
O total com desconto é R$ 4.51
